[Bug]: Intermittent failures in `reorganize_pages_spec` if text layer rendering is delayed

### Attach (recommended) or Link to PDF file

N/A

### Web browser and its version

Firefox 150.0.1

### Operating system and its version

Arch Linux

### PDF.js version

5.7.284 (and master)

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1. Apply the following diff locally:

```diff
diff --git a/test/integration/reorganize_pages_spec.mjs b/test/integration/reorganize_pages_spec.mjs
index 95c7de93c..e22145864 100644
--- a/test/integration/reorganize_pages_spec.mjs
+++ b/test/integration/reorganize_pages_spec.mjs
@@ -95,6 +95,7 @@ async function waitForHavingContents(page, expected) {
       for (const textLayer of document.querySelectorAll(".textLayer")) {
         buffer.push(parseInt(textLayer.textContent.trim(), 10));
       }
+      console.log(buffer);
       return ex.length === buffer.length && ex.every((v, i) => v === buffer[i]);
     },
     {},
@@ -1165,7 +1166,7 @@ describe("Reorganize Pages View", () => {
       await closePages(pages);
     });
 
-    it("should cut pages with Ctrl+X and paste them", async () => {
+    fit("should cut pages with Ctrl+X and paste them", async () => {
       await Promise.all(
         pages.map(async ([browserName, page]) => {
           await waitForThumbnailVisible(page, 1);
diff --git a/web/text_layer_builder.js b/web/text_layer_builder.js
index a94bbb16e..87005d542 100644
--- a/web/text_layer_builder.js
+++ b/web/text_layer_builder.js
@@ -101,6 +101,7 @@ class TextLayerBuilder {
       return;
     }
 
+    await new Promise(r => setTimeout(r, 200));
     this.cancel();
     this.#textLayer = new TextLayer({
       textContentSource: this.pdfPage.streamTextContent(
```

2. Run `npx gulp integrationtest --noChrome`.
3. Notice that the `should cut pages with Ctrl+X and paste them` integration test fails consistently with this traceback:

```
1) Reorganize Pages View Keyboard shortcuts for cut and copy (bug 2018139) should cut pages with Ctrl+X and paste them
  Message:
    Error: Waiting failed
  Stack:
    error properties: Object({ cause: ProtocolError: script.callFunction timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed. })
        at <instance_members_initializer> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/common/WaitTask.js:21:21)
        at new WaitTask (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/common/WaitTask.js:27:16)
        at BidiFrameRealm.waitForFunction (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/api/Realm.js:49:26)
        at BidiFrame.waitForFunction (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/api/Frame.js:580:43)
        at BidiFrame.<anonymous> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:101:27)
        at BidiPage.waitForFunction (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/api/Page.js:1447:37)
        at waitForHavingContents (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration/reorganize_pages_spec.mjs:92:15)
        at async file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration/reorganize_pages_spec.mjs:1190:11
        at async Promise.all (index 0)
        at async UserContext.<anonymous> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration/reorganize_pages_spec.mjs:1170:7)
    Caused by: ProtocolError: script.callFunction timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.
        at <instance_members_initializer> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:99:14)
        at new Callback (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:103:16)
        at CallbackRegistry.create (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:22:26)
        at BidiConnection.send (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Connection.js:70:32)
        at Session.send (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/core/Session.js:134:42)
        at Session.<anonymous> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:101:27)
        at WindowRealm.callFunction (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/core/Realm.js:92:51)
        at WindowRealm.<anonymous> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:101:27)
        at #evaluate (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Realm.js:136:42)
        at BidiFrameRealm.evaluateHandle (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/node_modules/puppeteer-core/lib/esm/puppeteer/bidi/Realm.js:103:36)
```

### What is the expected behavior?

The integration test should not be influenced by a delay in text layer rendering.

### What went wrong?

The integration test fails if a delay in text rendering is introduced.

### Link to a viewer

_No response_

### Additional context

This particular test failed a couple of times on GitHub Actions now, with https://github.com/mozilla/pdf.js/actions/runs/25277692228/job/74110020727#step:7:1099 being the latest occurrence.

The problem is not limited to this test: the introduced delay makes ~10 tests in the `reorganize_pages_spec` fail. There is a common root cause: if the delay is there then all tests hang in `waitForHavingContents`, and inspecting the buffer there with the attached diff shows that pages that have been cut/pasted have _duplicate_ text layer contents. For this particular test:

<img width="367" height="27" alt="Image" src="https://github.com/user-attachments/assets/2d72055a-2ea8-4c96-9353-f83c78e708c5" />

Note the `1212` value at the text, which is the literal value of the page's text layer's `textContent` field that this helper function enumerates. This suggests that for e.g. cut operations we can hit a state where the text layer contents are incorrect, and delaying the text layer is a reproducer for that, but I haven't found out how that can happen. It looks like for page editing operations we may be re-using the original page view, which in turn somehow renders the text content again and appends that to the already-existing text content.

@calixteman With this reproducer, does this ring a bell for you as to what could cause this?

--- a/web/text_layer_builder.js
+++ b/web/text_layer_builder.js
@@ -155,6 +155,8 @@ class TextLayerBuilder {
   cancel() {
     this.#textLayer?.cancel();
     this.#textLayer = null;
+    this.#renderingDone = false;
+    this.div.replaceChildren();
 
     this.highlighter?.disable();
     this.accessibilityManager?.disable();

